### DISCUSSION TITLE
How to filter node paths by timestamp when sampling?

### DISCUSSION THREAD

**Discussion by arrigonialberto86** [pyg-team/pytorch_geometric#3424] [Q&A] [answered]
I am working on a fraud-detection project using an heterogenous graph of different entities to classify target nodes as **fraud / no-fraud**. Some of these entities are introduced into the graph at different times (or they are otherwise removed), creating a temporal dependency that in relational settings I usually portray as a '_validity window_'.

Is there a way to 'mask' nodes that have not created yet according to timestamp? Do I need to create a multitude of graphs to account for every time window (e.g. month)?

My plan is to use `NeighborLoader` in conjunction with `SAGEConv` and `to_hetero`, but I came across this issue which does not seem to have a simple solution (plus, `relative temporal encoding` does not filter nodes, simply adds a _sequentiality_ information layer)

**Comment by rusty1s** [comment, marked as answer]
This is currently not supported, but we have plans to integrate temporal samplers in the near future. One workaround is to filter out edges returned by the sampler in which the source node is younger than a given target node:
```python
row, col = batch.edge_index
mask = batch.t[row] <= batch.t[col]
batch.edge_index = batch.edge_index[:, mask]
```

**Reply by arrigonialberto86** [reply]
Thanks, that would do it! Also, nice to know you have plans to integrate temp. samplers

**Reply by YoavLotem** [reply]
Hey!
Is the temporal sampler available?

**Reply by rusty1s** [reply]
Yes, it is available in the most recent torch-sparse release.

**Reply by YoavLotem** [reply]
Thanks for your quick response!
Could you please provide a link for a documentation or example?

**Reply by YoavLotem** [reply]
Found it on class NeighborLoader : )
https://pytorch-geometric.readthedocs.io/en/latest/modules/loader.html
Thanks!

**Reply by YoavLotem** [reply]
Can the sampler filter edges that are "older" than a specified center edge?
I'm working on an edge classification problem with temporal constraints.

**Reply by rusty1s** [reply]
Currently, the loader only support node-level time information. Sorry!
